chirp/drivers/tdh8.py

```diff
@@ -0,0 +1,2084 @@
+                                lenth_val += 1
+                        self._memobj.group6.group6 = list_val
+
+                    elif setting == 'group7' and element.value.get_mutable():
+                        val = str(element.value)
+                        list_val = []
+                        lenth_val = 0
+                        while lenth_val < (len(val)):
+                            if val[lenth_val] != ' ':
+                                list_val.append(int(val[lenth_val], 16))
+                                lenth_val += 1
+                            else:
+                                list_val.append(0xFF)
+                                lenth_val += 1
+                        self._memobj.group7.group7 = list_val
+
+                    elif setting == 'group8' and element.value.get_mutable():
+                        val = str(element.value)
+                        list_val = []
+                        lenth_val = 0
+                        while lenth_val < (len(val)):
+                            if val[lenth_val] != ' ':
+                                list_val.append(int(val[lenth_val], 16))
+                                lenth_val += 1
+                            else:
+                                list_val.append(0xFF)
+                                lenth_val += 1
+                        self._memobj.group8.group8 = list_val
+                    elif element.value.get_mutable():
+                        setattr(obj, setting, element.value)
+                except Exception:
+                    LOG.debug(element.get_name())
+                    raise
+
+    def _set_fm_preset(self, settings):
+        for element in settings:
+            try:
+                val = element.value
+                if self._memobj.fm_presets <= 108.0 * 10 - 650:
+                    value = int(val.get_value() * 10 - 650)
+                else:
+                    value = int(val.get_value() * 10)
+                LOG.debug("Setting fm_presets = %s" % (value))
+                self._memobj.fm_presets = value
+            except Exception:
+                LOG.debug(element.get_name())
+                raise
+
+
+@directory.register
+class TDH8_HAM(TDH8):
+    VENDOR = "TIDRADIO"
+    MODEL = "TD-H8-HAM"
+    ident_mode = b'P31185\xff\xff'
+    _ham = True
+    _vhf_range = (144000000, 149000000)
+    _uhf_range = (420000000, 451000000)
+
+
+@directory.register
+class TDH8_GMRS(TDH8):
+    VENDOR = "TIDRADIO"
+    MODEL = "TD-H8-GMRS"
+    ident_mode = b'P31184\xff\xff'
+    _gmrs = True
+    _vhf_range = (136000000, 175000000)
+    _uhf_range = (400000000, 521000000)
+
+    def validate_memory(self, mem):
+        msgs = super().validate_memory(mem)
+        if 31 <= mem.number <= 54 and mem.freq not in GMRS_FREQS:
+            msgs.append(chirp_common.ValidationError(
+                "The frequency in channels 31-54 must be between"
+                "462.55000-462.72500 in 0.025 increments."))
+        return msgs
```

Comment:
One other thing to check in here is that the duplex (and maybe offset) are right. If I choose duplex="-" for one of the GMRS-only channels, it flips back to + (which is good) but I get no warning about it. Just check here and add a warning (not error) so that the user will see.

Also, I can set offset=3MHz on GMRS channels, perhaps you want to enforce that it's always either simplex or +5MHz on those?